packages/token-translator/src/index.ts

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env node
+
+import path from "path";
+import { execSync } from "child_process";
+import fs from "fs";
+import fsPromises from "fs/promises";
+
+import postcss from "postcss";
+
+import postcssPlugin from "./postcss-token-translator";
+import { fileReplacer } from "./file-replacer";
+
+const cwd = process.cwd();
+
+/**
+ * Поиск файлов с необходимыми расширениями
+ */
+function searchFiles(startPath: string, extNames: string[]): string[] {
+  if (
+    ["node_modules"].includes(path.basename(startPath)) || // Всегда игнорируем node_modules
+    !fs.existsSync(startPath)
+  ) {
+    return [];
+  }
+
+  const findFiles: string[] = [];
+
+  fs.readdirSync(startPath).forEach((filename) => {
+    const pathToFile = path.join(startPath, filename);
+
+    const stat = fs.lstatSync(pathToFile);
+    if (stat.isDirectory()) {
+      findFiles.push(...searchFiles(pathToFile, extNames));
+    } else if (
+      extNames.includes(path.extname(pathToFile).substring(1)) &&
+      !stat.isSymbolicLink()
+    ) {
+      findFiles.push(pathToFile);
+    }
```

Comment:
Чтобы два раза расширение не получать, можно один раз здесь вычислить

```suggestion
    if (stat.isDirectory()) {
      findFiles.push(...searchFiles(pathToFile, extNames));
      return;
    }
    const extName = path.extname(pathToFile).substring(1);
    if (extNames.includes(extName) && !stat.isSymbolicLink()) {
      findFiles.push({
        path: pathToFile,
        ext: extName,
      });
    }
```

Вообще ещё можно было бы в одну итерацию это сделать если второй аргумент сделать парсером:

```tsx
function searchFiles<T>(startPath: string, parser: (pathToFile: string) => T): T[] {}
```

Reply:
При условии, что избавимся от `gitignore()` (см. комментарий про `gitinore()`)